oslc_prolog/lib/oslc_shape.pl

```diff
@@ -51,14 +51,16 @@
 check_value_type(_, _, [], _) :- !.
 
 check_value_type(IRI, PropertyResource, [V|T], Type) :-
-  once(rdf(PropertyResource, oslc:valueType, Type)),
-  once((
-    check_value(IRI, PropertyResource, V, Type),
-    check_allowed_values(IRI, PropertyResource, V, Type)
-  ; rdf(PropertyResource, oslc:propertyDefinition, PropertyDefinition),
-    oslc_error("Property [~w] of resource [~w] must be of type [~w]", [PropertyDefinition, IRI, Type])
-  )),
-  check_value_type(IRI, PropertyResource, T, Type).
+  (once(rdf(PropertyResource, oslc:valueType, Type))
+  -> once((
```

Comment:
FIX: if the `oslc:valueType` property is omitted, the shape should not do any type checking.